provider/kubernetes/kubernetes.go

```diff
@@ -159,13 +164,35 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 				default:
 					log.Warnf("Unknown value of %s for traefik.frontend.passHostHeader, falling back to %s", passHostHeaderAnnotation, PassHostHeader)
 				}
+				if realm := i.Annotations["ingress.kubernetes.io/auth-realm"]; realm != "" && realm != traefikDefaultRealm {
+					return nil, errors.New("no realm customization supported")
+				}
+
+				var basicAuthCreds []string
+				if authType := i.Annotations["ingress.kubernetes.io/auth-type"]; strings.ToLower(authType) == "basic" {
+					authSecret := i.Annotations["ingress.kubernetes.io/auth-secret"]
+					if authSecret == "" {
+						return nil, errors.New("auth-secret annotation must be set")
+					}
+					var err error
+					basicAuthCreds, err = loadAuthCredentials(i.Namespace, authSecret, k8sClient)
+					if err != nil {
+						return nil, err
+					}
+					if len(basicAuthCreds) == 0 {
+						return nil, errors.New("secet file without credentials")
+					}
+				} else if authType != "" {
+					return nil, fmt.Errorf("unsupported auth-type: %q", authType)
+				}
```

Comment:
Could we please put the entire block in a separate function?

`loadIngresses` is already pretty big.